eks-hybrid-iam-ra-setup.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+set -eux -o pipefail
+
+declare -r SECRETS_DIR="/.bottlerocket/rootfs/root/.aws"
+DRY_RUN="false"
+
+set +x
+for opt in "$@"; do
+    optarg="$(expr "${opt}" : '[^ =]*[= ]\(.*\)')"
+    case "${opt}" in
+       --certificate=*) NODE_CERT_DATA="${optarg}" ;;
+       --key=*) NODE_KEY_DATA="${optarg}" ;;
+       --dry-run=*) DRY_RUN="${optarg}" ;;
+    esac
+done
+set -x
+
+set +x
+if [ -z "$NODE_CERT_DATA" ]; then
+    echo "Unable to retrieve certificate data for IAM-RA"
+    echo "Please provide certificate contents as --certificate=<Certificate>"
+    exit 1
+fi
+
+if [ -z "$NODE_KEY_DATA" ]; then
+    echo "Unable to retrieve private key data for IAM-RA"
+    echo "Please provide private key contents as --key=<Key>"
+    exit 1
+fi
+set -x
+
+if ! [ -d "$SECRETS_DIR"]; then
+    echo "Error: Directory $SECRETS_DIR is missing"
+    exit 1
+fi
+
+if ! ls -dZ "$SECRETS_DIR" 2>/dev/null | grep -q ":secret_t:"; then
+    echo "Error: Directory $SECRETS_DIR is not labeled with secret_t"
+    exit 1
+fi
+
+set +x
+cat << EOF > "${SECRETS_DIR}/node.crt"
+${NODE_CERT_DATA}
+EOF
+
+cat << EOF > "${SECRETS_DIR}/node.key"
+${NODE_KEY_DATA}
+EOF
+set -x
+
+if $DRY_RUN; then
+    echo "DRY_RUN is set to true. Exiting without making changes."
+    exit 0
+fi
+
+variant_id="$(apiclient get os.variant_id | jq -r ".os.\"variant_id\"")"
+version_id="$(apiclient get os.version_id | jq -r ".os.\"version_id\"")"
```

Comment:
nit: can remove unnecessary escaping
```suggestion
variant_id="$(apiclient get os.variant_id | jq -r '.os.variant_id')"
version_id="$(apiclient get os.version_id | jq -r '.os.version_id')"
```